### PR TITLE
Board: Fix push/pull activity

### LINKED ISSUE #1519: [Bug] Card lables: Adding/removing label from card is not displayed in history (could be because of $push/$pull operations)

### PR DESCRIPTION
**Pull request by Dvinyanin**
Signed-off-by: Dvinyanin Alexandr <dvinyanin.alexandr@gmail.com>

### CODE REVIEW

File: plugins/activity-resources/src/components/utils.ts [hunk 1]
```diff
@@ -164,11 +166,30 @@ function getHiddenAttrs (client: TxOperations, _class: Ref<Class<Doc>>): Set<str
...
+    added: utx.$push?.[m.key],
+    removed: utx.$pull?.[m.key]
+  }
+  if (value.set !== undefined) {
+    value.set = await getRealValue(value.set)
   }
-  return val
+  if (value.added !== undefined) {
+    value.added = Array.isArray(value.added.$each)
+      ? (value.added.$each as any[]).map(getRealValue)
```
Comment (haiodo) [inline plugins/activity-resources/src/components/utils.ts]: Better to findAll with {_id: {$In: array}}

File: plugins/activity-resources/src/components/utils.ts [hunk 2]
```diff
@@ -164,11 +166,30 @@ function getHiddenAttrs (client: TxOperations, _class: Ref<Class<Doc>>): Set<str
...
-  const val = utx[m.key]
-
-  if (client.getHierarchy().isDerived(m._class, core.class.Doc) && typeof val === 'string') {
-    // We have an reference, we need to find a real object to pass for presenter
-    return await client.findOne(m._class, { _id: val as Ref<Doc> })
+  async function getRealValue (value: any): Promise<any> {
+    if (client.getHierarchy().isDerived(m._class, core.class.Doc) && typeof value === 'string') {
+      // We have an reference, we need to find a real object to pass for presenter
+      return await client.findOne(m._class, { _id: value as Ref<Doc> })
+    }
```
Comment (haiodo) [inline plugins/activity-resources/src/components/utils.ts]: value should be returned if not derived from Doc

Comment (Dvinyanin) [reply plugins/activity-resources/src/components/utils.ts]: Fixed

File: plugins/activity-resources/src/components/utils.ts (hunk 1 above)
Comment (Dvinyanin) [reply plugins/activity-resources/src/components/utils.ts]: Done

File: plugins/activity-resources/src/components/utils.ts [hunk 3]
```diff
@@ -164,11 +166,26 @@ function getHiddenAttrs (client: TxOperations, _class: Ref<Class<Doc>>): Set<str
...
+    removed: utx.$pull?.[m.key]
+  }
+  if (value.set !== undefined) {
+    ;[value.set] = await getAllRealValues([value.set])
   }
-  return val
+  if (value.added !== undefined) {
+    value.added = await getAllRealValues(Array.isArray(value.added.$each) ? value.added.$each : [value.added])
+  }
+  if (value.removed !== undefined) {
```
Comment (haiodo) [inline plugins/activity-resources/src/components/utils.ts]: Removed will not be located from snapshot storage and need to be restored from transactions. Please see how document it restored from transactions in activity.

Comment (Dvinyanin) [reply plugins/activity-resources/src/components/utils.ts]: It is only for `$pull` operation in `TxUpdateDoc`
But if `attachedDoc` was deleted, presenter will not be shown for all `txes`

**Review by haiodo — APPROVED** [review]

**Pull request merged**